template/utils/config.py

```diff
@@ -64,7 +64,9 @@ def add_args(cls, parser):
     # Netuid Arg: The netuid of the subnet to connect to.
     parser.add_argument("--netuid", type=int, help="Subnet netuid", default=1)
 
-    neuron_type = "validator" if issubclass(cls, BaseValidatorNeuron) else "miner"
```

Comment:
I have a WIP fix for the neuron type detection, but if it gives us pain let's just reintroduce config at the level of `BaseMinerNeuron` and `BaseValidatorNeuron`